### PR TITLE
Add Javadoc about path patterns

### PR DESCRIPTION
**Pull request by hyangtack**
Motivation:

I've missed adding Javadoc before merging the following PR:
https://github.com/line/armeria/pull/3562

Modifications:

- Add Javadoc to `AbstractBindingBuilder#exclude` and `RouteBuilder#exclude` methods.

### CODE REVIEW

**Review by ikhoon — APPROVED** [review]
Thanks! @hyangtack

**Review by minwoox — APPROVED** [review]
👍

File: core/src/main/java/com/linecorp/armeria/server/AbstractBindingBuilder.java [hunk 1]
```diff
@@ -409,6 +409,8 @@ public AbstractBindingBuilder addRoute(Route route) {
     /**
      * Adds a {@code pathPattern} that is supposed to be excluded from the {@link Route}s built by this
      * {@link AbstractBindingBuilder}.
+     * Please refer to the <a href="https://armeria.dev/docs/server-basics#path-patterns">Path patterns</a>
+     * in order to learn how to specify a path pattern.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/AbstractBindingBuilder.java:413]:
```suggestion
     * Please refer to <a href="https://armeria.dev/docs/server-basics#path-patterns">Path patterns</a>
     * to learn more about path pattern syntax.
```

File: core/src/main/java/com/linecorp/armeria/server/RouteBuilder.java [hunk 2]
```diff
@@ -435,6 +435,8 @@ RouteBuilder fallback(boolean isFallback) {
     /**
      * Adds a {@code pathPattern} that is supposed to be excluded from the {@link Route} built by this
      * {@link RouteBuilder}.
+     * Please refer to the <a href="https://armeria.dev/docs/server-basics#path-patterns">Path patterns</a>
+     * in order to learn how to specify a path pattern.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/RouteBuilder.java:439]:
```suggestion
     * Please refer to <a href="https://armeria.dev/docs/server-basics#path-patterns">Path patterns</a>
     * to learn more about path pattern syntax.
```

**Review by trustin — APPROVED** [review]

**Pull request merged**